File: src/baekjoon/step/greedy/PROB1541.py
# 입력받은 식에 괄호를 적절히 쳐서 식의 값을 최소로 만들기


# 문자열 입력받기
str = input().split('-')

# 숫자 저장하는 배열
numbers = []

# + 가 포함된 요소를 찾아서 나누고 덧셈 처리 하기
for i in str:
    number = 0
    temp = i.split('+')

    for j in temp:
        number += int(j)

    numbers.append(number)

# 완성된 리스트에서 0 번째 수에서 나머지 빼주기
result = numbers[0]

for i in range(1, len(numbers)):
    result -= numbers[i]

print(result)

File: src/baekjoon/level/silver/PROB1541.java
package silver;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class PROB1541 {
    static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

    static String[] input;
    static int min;

    public static void main(String[] args) throws IOException {
        input = br.readLine().split("-");

        for (int i = 0; i < input.length; i++) {
            int total = 0;
            String[] temp = input[i].split("\\+");

            for (int j = 0; j < temp.length; j++) {
                total += Integer.parseInt(temp[j]);
            }

            if (i == 0) min += total;
            else min -= total;
        }

        System.out.println(min);
    }
}
